[wasm] Use `--emit-symbol-map` to generate a `dotnet.js.symbols`

radical opened the pull request:
.. and package that with the runtime pack.

This file contains wasm function number to name mapping. This can be used to symbolicate traces with missing function names, like:

```
fail:     at <anonymous>:wasm-function[5721]:0x126d6e
fail:     at <anonymous>:wasm-function[258]:0x1c0b8
fail:     at <anonymous>:wasm-function[229]:0x1b6f1
```

.. can become:

```
fail:     at icu::number::impl::CompactData::~CompactData\28\29 (<anonymous>:wasm-function[5721]:0x126d6e)
fail:     at do_icall (<anonymous>:wasm-function[258]:0x1c0b8)
fail:     at do_icall_wrapper (<anonymous>:wasm-function[229]:0x1b6f1)
```

This is controlled by a new property - `$(WasmEmitSymbolMap)` which defaults to `false` right now, because the sdk needs a corresponding patch. This is set to `true` for tests though.

Support for doing this automatically will be added to XHarness in an upcoming PR.

radical committed:
[wasm] honor XunitShowProgress property

radical committed:
[wasm] avoid extra build when publishing

radical committed:
[wasm] Use --emit-symbol-map to generate a dotnet.js.symbols

.. and package it in the runtime pack. This can be disabled with
`$(WasmEmitSymbolMap)`.

radical committed:
wasm.proj: cleanup

radical commented:
/azp run runtime-wasm

pavelsavara commented:
This new file will need follow-up change in SDK. Last time I added file it was https://github.com/dotnet/sdk/pull/22920/files
Should it be included in `wwwroot` or not ? 
Should it be there also for `Release` configuration ?

radical committed:
Merge remote-tracking branch 'origin/main' into symbol-map

radical committed:
[wasm] WasmAppBuilder: Fix handling of ExtraFilesToDeploy

It incorrectly treated `TargetPath` metadata as a directory.

radical committed:
[wasm] Disable generation of symbol maps by default

- this is needed because the sdk needs to handle this file specifically,
  and would break tests.
- Once, the sdk gets the fix, then we can turn this on by default.

- For tests, turn it on by default

radical committed:
[wasm] Deploy dotnet.js.symbols with the tests

radical committed:
cleanup

radical committed:
[wasm] dotnet-install.sh logs errors from curl, while trying older urls,

.. don't let that fail the build.

radical commented:
/azp run runtime-wasm

radical commented:
Test failure is https://github.com/dotnet/runtime/issues/65335, unrelated to this PR.

radical committed:
Merge remote-tracking branch 'origin/main' into symbol-map

pavelsavara commented on eng/testing/tests.wasm.targets:
```diff
@@ -180,7 +187,8 @@
 
     <Copy SourceFiles="@(BundleFiles)"         DestinationFolder="$(BundleDir)%(TargetDir)" />
-    <Copy SourceFiles="@(_WasmVFSFilesToCopy)" DestinationFiles="$(BundleDir)\extraFiles\%(_WasmVFSFilesToCopy.TargetPath)" />
+    <Copy SourceFiles="@(_WasmVFSFilesToCopy)" DestinationFiles="$(BundleDir)\vfsFiles\%(_WasmVFSFilesToCopy.TargetPath)" />
```
This should be there probably only in Debug mode, otherwise the `mono-config.json` will always contain it and the runtime will download it. It would fail if it was missing. (I guess)

I suggest you create WBT for that

radical replied:
It won't go into mono-config.json . Adding a test is a good suggestion, thanks 👍

radical committed:
[wasm] Temporarily disable WasmEmitSymbolMap by default

.. because sdk needs a fix for blazor projects to handle
`dotnet.js.symbols`.

But explicitly enable this for tests.

radical commented:
/azp run runtime-wasm

kg commented on eng/testing/tests.wasm.targets:
```diff
@@ -9,11 +9,11 @@
     <!-- Run only if previous command succeeded -->
     <_ShellCommandSeparator Condition="'$(OS)' == 'Windows_NT'">&amp;&amp;</_ShellCommandSeparator>
     <_ShellCommandSeparator Condition="'$(OS)' != 'Windows_NT'">&amp;&amp;</_ShellCommandSeparator>
-    <WasmNativeStrip>false</WasmNativeStrip>
```
are there any potential problems caused by disabling strip - for example, does this make our tests less representative because now they won't surface bugs that only occur when stripping?

radical replied:
This is actually set earlier in the file, and it was redundant here, that's why I removed it. So, no effective change.

radical replied:
re:disabling-strip, yeah, that itself could maybe not be completely representative. Once the symbolication work is fully merged (another PR needed), we can maybe drop this, so it is the same as user builds.

kg commented on src/mono/sample/wasm/wasm.mk:
```diff
@@ -3,7 +3,7 @@ DOTNET=$(TOP)/dotnet.sh
 DOTNET_Q_ARGS=--nologo -v:q -consoleloggerparameters:NoSummary
 else
-DOTNET_Q_ARGS=--nologo
+DOTNET_Q_ARGS=--nologo -bl
```
we should set this on both arms or not at all imo

radical committed:
Address review feedback

radical commented:
/azp run runtime-wasm

radekdoulik commented on src/mono/wasm/wasm.proj:
```diff
@@ -162,10 +162,24 @@
       <CMakeConfigurationEmccFlags Condition="'$(Configuration)' == 'Release'">-Oz</CMakeConfigurationEmccFlags>
-      <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Debug'">$(CMakeConfigurationEmccFlags)</CMakeConfigurationLinkFlags>
+
+      <CMakeConfigurationLinkFlags Condition="'$(Configuration)' == 'Debug'"  >$(CMakeConfigurationEmccFlags)</CMakeConfigurationLinkFlags>
```
nit: excessive space

radical replied:
Just matching other `>`s in the block, to make it easier to read.

radical commented:
`System.Collections.Immutable.Tests` is failing due to https://github.com/dotnet/runtime/issues/65335 , unrelated to this PR.

radical commented:
The library test failures are https://github.com/dotnet/runtime/issues/65335 .
The AOT ones timed out waiting to get queued on CI.

radical committed:
Merge branch 'dotnet:main' into symbol-map

lewing reviewed (approved):
Commit when you are happy with the state

radical committed:
Merge branch 'dotnet:main' into symbol-map

radical commented:
/azp run runtime-wasm

radical commented:
The CoreCLR/win/arm64 issues are https://github.com/dotnet/runtime/issues/65818 .
The CoreCLR/win/x64 issue is https://github.com/dotnet/runtime/issues/65829 .
